go.mod

```
@@ -120,4 +120,4 @@ require (
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-replace github.com/distribution/distribution/v3 => github.com/openshift/docker-distribution/v3 v3.0.0-20240826125755-2799294adc70
+replace github.com/distribution/distribution/v3 => github.com/mpryc/docker-distribution/v3 v3.0.0-20240826125755-2799294adc70
```

pretty sure the hash has to match PR so the last part needs to have a substring from f30569206c423e99ffb1e900c4d05d5a221795b9

```suggestion
replace github.com/distribution/distribution/v3 => github.com/mpryc/docker-distribution/v3 testing_distribution_pr_44
```
then `go mod tidy` should work.

---

Nope,
```
go.mod:123: replace github.com/mpryc/docker-distribution/v3: version "testing_distribution_pr_44" invalid: unknown revision testing_distribution_pr_44
```

Now what I did was to point to my fork and my branch and then run go mod tidy, and this PR is a result of it:
```
$ git diff
diff --git a/go.mod b/go.mod
index 0972866f0..df349770d 100644
--- a/go.mod
+++ b/go.mod
@@ -120,4 +120,4 @@ require (
        sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-replace github.com/distribution/distribution/v3 => github.com/openshift/docker-distribution/v3 v3.0.0-20240826125755-2799294adc70
+replace github.com/distribution/distribution/v3 => github.com/mpryc/docker-distribution/v3 OCPBUGS-25981

$ go mod tidy

$ git diff
diff --git a/go.mod b/go.mod
index ee42883e9..0972866f0 100644
--- a/go.mod
+++ b/go.mod
@@ -120,4 +120,4 @@ require (
        sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-replace github.com/distribution/distribution/v3 => github.com/openshift/docker-distribution/v3 v3.0.0-20240826125755-2799294adc70
+replace github.com/distribution/distribution/v3 => github.com/mpryc/docker-distribution/v3 v3.0.0-20240826125755-2799294adc70
diff --git a/go.sum b/go.sum
index d9c0a4dca..4f6cd617d 100644
--- a/go.sum
+++ b/go.sum
@@ -213,6 +213,8 @@ github.com/modern-go/reflect2 v1.0.2 h1:xBagoLtFs94CBntxluKeaWgTMpvLxC4ur3nMaC9G
 github.com/modern-go/reflect2 v1.0.2/go.mod h1:yWuevngMOJpCy52FWWMvUC8ws7m/LJsjYzDa0/r8luk=
 github.com/morikuni/aec v1.0.0 h1:nP9CBfwrvYnBRgY6qfDQkygYDmYwOilePFkwzv4dU8A=
 github.com/morikuni/aec v1.0.0/go.mod h1:BbKIizmSmc5MMPqRYbxO4ZU0S0+P200+tUnFx7PXmsc=
+github.com/mpryc/docker-distribution/v3 v3.0.0-20240826125755-2799294adc70 h1:f2iylj3ku8w/yP/7Vfq6n5wB8xgUObl6nie0NEDR9lc=
+github.com/mpryc/docker-distribution/v3 v3.0.0-20240826125755-2799294adc70/go.mod h1:+fqBJ4vPYo4Uu1ZE4d+bUtTLRXfdSL3NvCZIZ9GHv58=
 github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822 h1:C3w9PqII01/Oq1c1nUAm88MOHcQC9l5mIlSMApZMrHA=
 github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822/go.mod h1:+n7T8mK8HuQTcFwEeznm/DIxMOiR9yIdICNftLE1DvQ=
 github.com/mwitkow/go-conntrack v0.0.0-20161129095857-cc309e4a2223/go.mod h1:qRWi+5nqEBWmkhHvq77mSJWrCKwh8bxhgT7d/eI7P4U=
@@ -230,8 +232,6 @@ github.com/openshift/api v0.0.0-20240613141850-76a71dac36a0 h1:Kn16YZDBGwetg+pMQ
 github.com/openshift/api v0.0.0-20240613141850-76a71dac36a0/go.mod h1:OOh6Qopf21pSzqNVCB5gomomBXb8o5sGKZxG2KNpaXM=
 github.com/openshift/client-go v0.0.0-20240528061634-b054aa794d87 h1:JtLhaGpSEconE+1IKmIgCOof/Len5ceG6H1pk43yv5U=
 github.com/openshift/client-go v0.0.0-20240528061634-b054aa794d87/go.mod h1:3IPD4U0qyovZS4EFady2kqY32m8lGcbs/Wx+yprg9z8=
-github.com/openshift/docker-distribution/v3 v3.0.0-20240826125755-2799294adc70 h1:wmX3vWvDiW+Sjsa4c17Dh2+ewmklPcftw/Ut3cfOUGM=
-github.com/openshift/docker-distribution/v3 v3.0.0-20240826125755-2799294adc70/go.mod h1:+fqBJ4vPYo4Uu1ZE4d+bUtTLRXfdSL3NvCZIZ9GHv58=
 github.com/openshift/library-go v0.0.0-20240607134135-aed018c215a1 h1:jLERUXwvYY9+9oCz66oQ/XTZzeyH8RmCpxiImYVYnmA=
 github.com/openshift/library-go v0.0.0-20240607134135-aed018c215a1/go.mod h1:PdASVamWinll2BPxiUpXajTwZxV8A1pQbWEsCN1od7I=
 github.com/pborman/uuid v1.2.1 h1:+ZZIw58t/ozdjRaXh/3awHfmWRbzYxJoAdNJxe/3pvw=
```

As you can see `go mod tidy` replaced pointer with it's own hashes/etc...